base/actions/actions.py

```diff
@@ -724,14 +724,66 @@ async def submit(
                         resp = await client.post(url, json=post_data, headers=headers)
                         # TODO: remove print
                         print(resp.content)
+                        json_resp = resp.json()
 
+                        # Get tbcheck ID
+                        tbcheck_id = json_resp.get("id")
+
+                        # Get template and user group arm
                         templates, group_arm = utils.get_display_message_template(resp)
 
+                        if not group_arm:
+                            templates = utils.get_risk_templates(risk, data)
+
+                        # Get clinic list for
+                        if group_arm == "planning_prompt":
+                            location = json_resp.get("location")
+                            longitude, latitude = utils.extract_location_long_lat(
+                                location, 2
+                            )
+
+                            if longitude and latitude:
+                                querystring = urlencode(
+                                    {"longitude": longitude, "latitude": latitude}
+                                )
+
+                                clinic_url = urljoin(
+                                    config.HEALTHCONNECT_URL,
+                                    "/v1/clinic_finder" f"?{querystring}",
+                                )
+
+                                nearest_clinic = await client.get(
+                                    clinic_url, headers=headers
+                                )
+
+                                clinic_list = ""
+                                original_clinic = []
+                                list_num = 0
+                                if nearest_clinic:
+                                    for clinic in nearest_clinic.json().get(
```

Comment:
I would clean some of this up, try building the clinic_list string in a utiil function maybe?